api/envoy/service/status/v3/csds.proto

```diff
@@ -124,10 +126,54 @@ message ClientConfig {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.service.status.v2.ClientConfig";
 
+  // GenericXdsConfig is a simpler and generic version of specifying
+  // XDS config status and the dump.
+  // [#next-free-field: 9]
+  message GenericXdsConfig {
+    // type_url represents the fully qualified name of xDS resource type
+    // like envoy.v3.Cluster, envoy.v3.ClusterLoadAssignment etc.
+    string type_url = 1;
+
+    // name of the xDS resource
+    string name = 2;
```

Comment:
Yes, it represents a single resource.

In practice, I suspect that most implementations will send all resources of the same type next to each other anyway, simply because that's the obvious way to implement CSDS.  But you're right that there is no requirement to do that here.

Keep in mind, though, that even if the server doesn't group those resources together, the client can always group them that way for display purposes.  I think the API and the UX can be handled independently.

Reply:
SG.